### PR TITLE
Add scale and background lines to Experiment and Analysis plots

### LINKED ISSUE #84: Plotting background and scale lines
In the experiment tab, it would be nice to have a horizontal line for the background and scale levels. I think this would help to estimate these values.

### PR DESCRIPTION
**Pull request by arm61**
This PR adds the ability to have dashed lines indicating the scale and background levels to the analysis and experiment tabs. This is helpful to allow the user to select accurate starting values for the scale and background levels. 

This will close #84

### CODE REVIEW

File: EasyReflectometryApp/Logic/Proxies/Simulation.py [hunk 1]
```diff
@@ -189,6 +189,8 @@ def _updateCalculatedData(self):
         sld_profile = self.parent._interface.sld_profile(to_use.uid)
 
         self.parent._plotting_1d_proxy.setCalculatedData(x, y)
+        self.parent._plotting_1d_proxy.setBackgroundData(x, np.ones_like(x) * to_use.background.raw_value)
+        self.parent._plotting_1d_proxy.setScaleData(x, np.ones_like(x) * to_use.scale.raw_value)
```
Comment (rozyczko) [inline EasyReflectometryApp/Logic/Proxies/Simulation.py:193]: minor nitpick but 
```
np.repeat( to_use.background.raw_value, (x))
```
is quicker and maybe even more readable
https://stackoverflow.com/questions/5891410/numpy-array-initialization-fill-with-identical-values

**Review by rozyczko — APPROVED** [review]
I don't see anything outrageous; code looks fine, does what it is supposed to.

File: EasyReflectometryApp/Logic/Proxies/Simulation.py (hunk 1 above)
Comment (arm61) [reply EasyReflectometryApp/Logic/Proxies/Simulation.py:193]: Woah, I have never heard of `np.repeat` 😱

File: EasyReflectometryApp/Gui/Logic/Plotting.js [hunk 2]
```diff
@@ -276,6 +282,38 @@ function bokehAddVisibleYAxis(chart, specs) {
...
+function bokehAddBackgroundDataToMainChart(data, specs) {
+    return [`main_source.data.x_bkg = [${data.background.x}]`,
+            `main_source.data.y_bkg = [${data.background.y}]`,
+
+            'const bkgLine = new Bokeh.Line({',
+            '    x: { field: "x_bkg" },',
+            '    y: { field: "y_bkg" },',
+            `    line_color: "${specs.backgroundLineColor}",`,
+            `    line_alpha: 0.5,`,
+            `    line_dash: [6, 2],`,
```
[the remark this answers is not in the document]
Comment (arm61) [reply EasyReflectometryApp/Gui/Logic/Plotting.js:294]: It was to make it visually distinct from the scale line which is also dashed.

File: EasyReflectometryApp/Logic/Proxies/Simulation.py (hunk 1 above)
Comment (arm61) [reply EasyReflectometryApp/Logic/Proxies/Simulation.py]: I changed to the fastest option of `a = np.empty(x.size); a.fill(b)`.

File: EasyReflectometryApp/Gui/Pages/Analysis/SideBarAdvanced.qml [hunk 3]
```diff
@@ -157,9 +157,25 @@ EaComponents.SideBarColumn {
...
             id: rq4
             checked: ExGlobals.Constants.proxy.simulation.plotRQ4
-            text: qsTr("Show R(q)q⁴")
+            text: qsTr("Show as R(q)q⁴")
             onToggled: ExGlobals.Constants.proxy.simulation.setPlotRQ4()
         }
+
+        EaElements.CheckBox {
+            topPadding: 0
+            id: scale
```
Comment (AndrewSazonov) [inline EasyReflectometryApp/Gui/Pages/Analysis/SideBarAdvanced.qml:166]: It seems that you are not referring to this element, so there is no need to specify its `id` explicitly. The same is true for `rq4` and `bkg`.

File: EasyReflectometryApp/Gui/Components/BasePlot.qml [hunk 4]
```diff
@@ -54,10 +60,14 @@ Rectangle {
     property color measuredAreaColor: measuredLineColor
     property color calculatedLineColor: EaStyle.Colors.chartForegrounds[1]
     property color sldLineColor: EaStyle.Colors.chartForegrounds[2]
+    property color backgroundLineColor: EaStyle.Colors.chartForegrounds[3]
+    property color scaleLineColor: EaStyle.Colors.chartForegrounds[4]
```
Comment (AndrewSazonov) [inline EasyReflectometryApp/Gui/Components/BasePlot.qml:64]: Is there item 4 in the `EaStyle.Colors.chartForegrounds` list?

**Review by AndrewSazonov — COMMENTED** [review]
Look okay. I added a few minor comments. Also, I have not found how to set a non-zero background. Does this only become available when loading experimental data? If so, should it be part of the simulation-only workflow?

File: EasyReflectometryApp/Gui/Components/BasePlot.qml (hunk 4 above)
Comment (arm61) [reply EasyReflectometryApp/Gui/Components/BasePlot.qml:64]: The lines look to be the same colour (but I can't tell the difference). It doesn't matter though cause they are visually distinct due to different dashing.

**Comment by arm61** [comment]
The background and scale are not parameters available in the sample tab. Only the experiment and analysis tabs, cause the sample data is "pure" i.e. no experimental contributions.

**Pull request merged**